mongodbforms/documents.py

```diff
@@ -129,6 +143,10 @@ def construct_instance(form, instance, fields=None, exclude=None):
                     list_field[i] = file_obj
                 except IndexError:
                     list_field.append(file_obj)
+
+            for idx in reversed(idx_to_pop):
+                list_field.pop(idx)
```

Comment:
No ambiguity, noone will wonder if you're looking for a `idx` item in the list itself.